source/fab/steps/grab.py

```diff
@@ -157,3 +155,40 @@ def run(self, artefact_store: Dict, config):
 
             r = remote.RemoteClient(self.src)
             r.export(str(config.source_root / self.dst_label), revision=self.revision, force=True)
+
+
+class GrabPreBuild(Step):
```

Comment:
We may want to think about alternate naming. So far "grab" has always been used to build a source tree for compilation. Here it's being used to manage internal state.

Reply:
Always up for those chats.